### PR TITLE
[ios] Migrate expo-localization to the new API

### PR DESCRIPTION
**Pull request by tsapeta**
# Why

Migrating more modules to the new Expo Modules API

# How

- Renamed pod name to `ExpoLocalization` and updated the podspec to use Swift
- Created `LocalizationModule.swift` and mirrored current functionality
- Renamed `unimodule.json` to `expo-module.config.json` and added the module class name there

# Test Plan

NCL and test-suite are passing

### CODE REVIEW

File: packages/expo-localization/ios/LocalizationModule.swift [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+      "currency": locale.currencyCode ?? "USD",
+      "decimalSeparator": locale.decimalSeparator ?? ".",
+      "digitGroupingSeparator": locale.groupingSeparator ?? ",",
+      "isoCurrencyCodes": Locale.isoCurrencyCodes,
+      "isMetric": locale.usesMetricSystem,
+      "isRTL": Locale.characterDirection(forLanguage: languageCode) == .rightToLeft,
+      "locale": languageIds.first,
+      "locales": languageIds,
+      "region": locale.regionCode ?? "US",
+      "timezone": TimeZone.current.identifier
```
Comment (EvanBacon) [inline packages/expo-localization/ios/LocalizationModule.swift:38]: Is this analogous to `[NSTimeZone localTimeZone].name`?

**Review by EvanBacon — APPROVED** [review]

File: packages/expo-localization/ios/LocalizationModule.swift (hunk 1 above)
Comment (tsapeta) [reply packages/expo-localization/ios/LocalizationModule.swift:38]: Yeah, it seems to return the same value but I didn't find any confirmation on the internet other than this https://stackoverflow.com/a/34537245/2406251

**Comment by mariomurrent-softwaresolutions** [comment]
I upgraded to Expo SDK 44 and now I'm getting the following error, when build via EAS:

` (node_modules/expo-localization/ios/LocalizationModule.swift:14:5)

  12 |     }
  13 | 
> 14 |     function("getLocalizationAsync") {
     |     ^ cannot find 'function' in scope
  15 |       return Self.getCurrentLocalization()
  16 |     }
  17 |   }`
  
  expo-localization version 12.0.0

**Comment by tsapeta** [comment]
@mariomurrent-softwaresolutions Can you run `yarn why expo-modules-core` and send the results here? I'm sure you use the old version 😞

**Comment by mariomurrent-softwaresolutions** [comment]
@tsapeta Sure => => Found "expo#expo-modules-core@0.6.3"

**Comment by tsapeta** [comment]
Interesting, `0.6.3` is correct for SDK44 🤔 Are you sure it's the only version it founds? I guess you may have duplicates somewhere and it could result in linking incorrect version.

**Comment by mariomurrent-softwaresolutions** [comment]
I can see some other version as peer dependencies in the yarn.lock

**Comment by tsapeta** [comment]
Ok, can you also send me your dependencies and their versions from package.json?

**Comment by mariomurrent-softwaresolutions** [comment]
I actually now did and expo install again for each of these modules and it seems to work now.
I'm just getting a different error while building - but this seems to be related to something else.

Thx for the help

**Comment by mariomurrent-softwaresolutions** [comment]
> Ok, can you also send me your dependencies and their versions from package.json?

I'm still using react-native-unimodules. 

Could this case the issue?

**Comment by tsapeta** [comment]
Yes, it could. It's deprecated now and we asked to remove it as part of upgrading to SDK43, here is the guide https://github.com/expo/fyi/blob/main/expo-modules-migration.md

**Comment by mariomurrent-softwaresolutions** [comment]
@tsapeta Does not seem to work - same error now.
I have the following dependencies:

` "dependencies": {
    "@expo/react-native-action-sheet": "^3.12.0",
    "@expo/samples": "3.0.3",
    "@expo/vector-icons": "^12.0.0",
    "@freakycoder/react-native-bounceable": "^0.2.4",
    "@freakycoder/react-native-single-select": "^0.3.8",
    "@freakycoder/react-native-text-area": "^0.1.5",
    "@freakycoder/react-native-text-input": "0.1.8",
    "@gapur/react-native-accordion": "^1.0.5",
    "@meecode/ng112-reg-js": "^1.0.4",
    "@react-native-async-storage/async-storage": "~1.15.0",
    "@react-native-community/masked-view": "0.1.10",
    "@react-native-community/netinfo": "7.1.3",
    "@react-native-picker/picker": "2.2.1",
    "@react-navigation/bottom-tabs": "^6.0.7",
    "@react-navigation/native": "^6.0.4",
    "@react-navigation/stack": "^6.0.9",
    "@sentry/react-native": "^3.2.8",
    "@unimodules/core": "~7.2.0",
    "@unimodules/react-native-adapter": "~6.5.0",
    "axios": "^0.23.0",
    "date-fns": "^2.22.1",
    "expo": "^44.0.0",
    "expo-analytics": "^1.0.16",
    "expo-analytics-amplitude": "~11.1.0",
    "expo-app-loading": "~1.3.0",
    "expo-application": "~4.0.1",
    "expo-av": "~10.2.0",
    "expo-battery": "~6.1.0",
    "expo-cellular": "~4.1.0",
    "expo-clipboard": "~2.1.0",
    "expo-constants": "~13.0.0",
    "expo-crypto": "~10.1.1",
    "expo-dev-client": "^0.7.2",
    "expo-device": "~4.1.0",
    "expo-font": "~10.0.4",
    "expo-google-sign-in": "~10.1.0",
    "expo-keep-awake": "~10.0.1",
    "expo-linear-gradient": "~11.0.0",
    "expo-linking": "~3.0.0",
    "expo-localization": "~11.0.0",
    "expo-location": "~14.0.1",
    "expo-modules-autolinking": "^0.3.4",
    "expo-navigation-bar": "~1.1.1",
    "expo-network": "~4.1.0",
    "expo-permissions": "~13.1.0",
    "expo-secure-store": "~11.1.0",
    "expo-splash-screen": "~0.14.0",
    "expo-status-bar": "~1.2.0",
    "expo-store-review": "~5.1.0",
    "expo-system-ui": "~1.0.0",
    "expo-task-manager": "~10.1.0",
    "expo-updates": "~0.11.2",
    "expo-web-browser": "~10.1.0",
    "i18n-js": "^3.8.0",
    "jssip": "^3.8.0",
    "jssip-node-websocket": "^3.0.0",
    "lodash": "^4.17.21",
    "lottie-ios": "^3.2.0",
    "lottie-react-native": "5.0.1",
    "native-base": "^2.15.2",
    "next-fonts": "^1.5.1",
    "ng112-js": "^1.5.2",
    "prop-types": "^15.7.2",
    "pubsub-js": "^1.9.3",
    "react": "17.0.1",
    "react-dom": "17.0.1",
    "react-native": "0.64.3",
    "react-native-app-intro-slider": "^4.0.4",
    "react-native-bouncy-checkbox": "^2.1.3",
    "react-native-confirmation-code-field": "^7.1.0",
    "react-native-expo-viewport-units": "0.0.8",
    "react-native-form": "^2.1.3",
    "react-native-gesture-handler": "~2.1.0",
    "react-native-gifted-chat": "^0.16.3",
    "react-native-google-places-autocomplete": "^2.4.1",
    "react-native-health": "^1.11.0",
    "react-native-htmlview": "^0.16.0",
    "react-native-iphone-x-helper": "^1.3.1",
    "react-native-json-tree": "^1.3.0",
    "react-native-keyboard-aware-scroll-view": "^0.9.4",
    "react-native-loading-spinner-overlay": "^2.0.0",
    "react-native-logs": "^3.0.4",
    "react-native-maps": "0.29.4",
    "react-native-modalize": "^2.0.8",
    "react-native-modals": "^0.22.3",
    "react-native-offline": "^6.0.0",
    "react-native-popover-tooltip": "^1.1.4",
    "react-native-portalize": "^1.0.7",
    "react-native-prompt-android": "^1.1.0",
    "react-native-reanimated": "~2.3.1",
    "react-native-responsive-dimensions": "^3.1.1",
    "react-native-safe-area-context": "3.3.2",
    "react-native-screens": "~3.10.1",
    "react-native-simple-crypto": "^0.2.15",
    "react-native-size-matters": "^0.4.0",
    "react-native-spinkit": "^1.5.1",
    "react-native-step-indicator": "1.0.3",
    "react-native-super-grid": "^4.0.3",
    "react-native-uuid": "^2.0.1",
    "react-native-vector-icons": "^8.1.0",
    "react-native-web": "0.17.1",
    "react-native-webview": "11.15.0",
    "react-native-whatsnew-kit": "0.0.2",
    "react-redux": "^7.2.4",
    "redux": "^4.1.0",
    "redux-thunk": "^2.3.0",
    "rxjs": "^7.2.0",
    "sentry-expo": "^4.0.0",
    "sharp": "^0.29.3",
    "toggle-switch-react-native": "^3.2.0",
    "uuid": "^3.3.2",
    "wretch": "^1.7.5",
    "xmldom": "^0.4.0"
  },`

**Comment by tsapeta** [comment]
Looks like you should also remove `@unimodules/core` and `@unimodules/react-native-adapter`

**Comment by mariomurrent-softwaresolutions** [comment]
Okay, Ill try that. Thx

**Comment by mariomurrent-softwaresolutions** [comment]
> Yes, it could. It's deprecated now and we asked to remove it as part of upgrading to SDK43, here is the guide https://github.com/expo/fyi/blob/main/expo-modules-migration.md

But this means I need to use expo prebuild, right? This generates and iOS and android folder? But eas build does work right?

**Comment by mariomurrent-softwaresolutions** [comment]
> Looks like you should also remove `@unimodules/core` and `@unimodules/react-native-adapter`

I removed unimodales and adapter, but this did not help .. same error

**Comment by mariomurrent-softwaresolutions** [comment]
> @mariomurrent-softwaresolutions Can you run `yarn why expo-modules-core` and send the results here? I'm sure you use the old version 😞

But I don't need this for the managed workflow right?

**Pull request merged**